File: backend/src/main/java/unpsjb/labprog/backend/model/DTO/PublicacionesDTO.java
package unpsjb.labprog.backend.model.DTO;

import lombok.Getter;
import lombok.Setter;
import unpsjb.labprog.backend.model.Comunidad;
import unpsjb.labprog.backend.model.Publicacion;
import unpsjb.labprog.backend.model.Usuario;

@Getter
@Setter
public class PublicacionesDTO {
    public Publicacion publicacion;
    public Usuario creador;
    public Long likes;
    public boolean estaLikeada;
    public Comunidad comunidad;
}

File: frontend/cli/src/app/home/PublicacionesDto.ts
import { Comunidad } from "../comunidades/comunidad";
import { Publicacion } from "../publicaciones/publicacion";
import { Usuario } from "../usuarios/usuario";

export interface PublicacionDto {
    publicacion: Publicacion;
    creador: Usuario;
    likes: number;
    estaLikeada: boolean;
    comunidad: Comunidad;
}